compiler/src/iree/compiler/Dialect/Flow/Transforms/SetEncoding.cpp

```diff
@@ -335,33 +317,26 @@ struct FoldFillWithSetEncoding
   }
 };
 
-struct SetEncodingPass : public SetEncodingBase<SetEncodingPass> {
-  void getDependentDialects(DialectRegistry &registry) const override {
-    registry.insert<IREE::Encoding::IREEEncodingDialect>();
-  }
-  explicit SetEncodingPass(int64_t factor) { this->padFactor.setValue(factor); }
-
-  void runOnOperation() override;
-};
-} // namespace
+class SetEncodingPass : public impl::SetEncodingPassBase<SetEncodingPass> {
 
-void SetEncodingPass::runOnOperation() {
-  MLIRContext *context = &getContext();
-  {
-    RewritePatternSet patterns(context);
-    patterns.insert<setContractionOpEncoding>(context, padFactor);
-    linalg::FillOp::getCanonicalizationPatterns(patterns, context);
-    patterns.insert<FoldFillWithSetEncoding>(context);
-    memref::populateResolveRankedShapedTypeResultDimsPatterns(patterns);
-    if (failed(applyPatternsAndFoldGreedily(getOperation(),
-                                            std::move(patterns)))) {
-      return signalPassFailure();
+public:
+  using Base::Base;
+
+  void runOnOperation() override {
+    MLIRContext *context = &getContext();
+    {
+      RewritePatternSet patterns(context);
+      patterns.insert<setContractionOpEncoding>(context, padFactor);
+      linalg::FillOp::getCanonicalizationPatterns(patterns, context);
+      patterns.insert<FoldFillWithSetEncoding>(context);
+      memref::populateResolveRankedShapedTypeResultDimsPatterns(patterns);
+      if (failed(applyPatternsAndFoldGreedily(getOperation(),
+                                              std::move(patterns)))) {
+        return signalPassFailure();
+      }
     }
```

Comment:
nit: I think we can remove one level of nesting here.

```suggestion
    RewritePatternSet patterns(context);
    patterns.insert<setContractionOpEncoding>(context, padFactor);
    linalg::FillOp::getCanonicalizationPatterns(patterns, context);
    patterns.insert<FoldFillWithSetEncoding>(context);
    memref::populateResolveRankedShapedTypeResultDimsPatterns(patterns);
    if (failed(applyPatternsAndFoldGreedily(getOperation(),
                                            std::move(patterns)))) {
      return signalPassFailure();
    }
```